### ISSUE TITLE
Not able to run OPOLO with Humanoid-v2

### ISSUE THREAD

**Issue by brokenBrain** [illidanlab/opolo-code#2]
I am able to run OPOLO with Walker2d and HalfCheetah, but for some reason the code fails when I try to run it with Humanoid. There's nothing wrong with the Mujoco installation. I'm able to train Humanoid when I'm using your expert trajectories and my algorithm. Looking at the OPOLO training logs, it looks like the gradients in the discriminator are exploding.

**Comment by zhuangdizhu** [comment]
Could you try to: 1) scale down the generated reward & loss, 2) clip the discriminator gradient to a certain range, or 3) try other random seeds ?

**Comment by brokenBrain** [comment]
I've tried multiple random seeds. I doubt 1) will work since it seems to be a problem in the discriminator. I'll look into 2). But I'm curious to know how you were able to get Humanoid working. I'm unable to find gradient clipping code in the repo. Did you have to use gradient clipping as well?

**Comment by brokenBrain** [comment]
I tried gradient clipping with multiple values and it didn't work. I also tried setting the gradient penalty coefficient to a value other than 10 --- that didn't work either. However, when I set the gradient penalty coefficient to 0, the code didn't error out. Training proceeded without a hitch. But the discriminator just ended up overfitting on the data.

**Comment by zhuangdizhu** [comment]
By default, the gradient penalty coefficient is set to 0 for all environments, see this [source](https://github.com/illidanlab/opolo-code/blob/d104f10aec1cc9ad7d2f91431d8e18ed1de7a402/opolo-baselines/stable_baselines/td3/td3opolo.py#L134) code. 

You might need to fine-tune some hyperparameters, including the buffer_size/learning_rate/ pi_reg_coeff.

**Comment by brokenBrain** [comment]
Thanks for getting back. The source code that you have pointed to only evaluates the gradient penalty for logging purposes. [Here](https://github.com/illidanlab/opolo-code/blob/main/opolo-baselines/stable_baselines/gail/gaifo.py), you will see that the gradient penalty is always on with a default coefficient of 10. I'd still appreciate guidance on how you got Humanoid working with the code base. What setting of parameters did you use?

**Comment by zhuangdizhu** [comment]
I use the hyper-parameters set in `https://github.com/illidanlab/opolo-code/blob/main/opolo-baselines/hyperparams/opolo.yml`, see the settings for `Humanoid-v2` [here](https://github.com/illidanlab/opolo-code/blob/d104f10aec1cc9ad7d2f91431d8e18ed1de7a402/opolo-baselines/hyperparams/opolo.yml#L129).
 
I did not reproduce the gradient exploration issue that occurred to you using the default hyper-parameters, with gradcoeff=10 set for GAIfO discriminator. I run it on a macOS this time (due to my expired mujoco license), with gym = 0.18.0, Tensorflow = 1.15.0, mujoco-py  = 2.0.2.13.  

Btw, what command did you use to run the opolo ?  Please include the tag`--task td3-opolo-idm-decay-reg` when running the algorithm.
